pkg/kubectl/cmd/util/helpers.go

```diff
@@ -428,3 +429,39 @@ func UpdateObject(info *resource.Info, updateFn func(runtime.Object) error) (run
 
 	return info.Object, nil
 }
+
+// isClusterService checks the resource's label to see if it's a cluster service (add-on)
+func isClusterService(info *resource.Info) (bool, error) {
```

Comment:
We need write protection in a bunch of places with different resources so making this generic is desirable.